Update third-party chart versions

NicolasT opened the pull request:
- Basic test for `metrics-server`
- Updates of various chart versions
- Update Elasticsearch to use `RollingUpdate` deployments

NicolasT committed:
tests: Add basic `metrics-server` test

NicolasT committed:
kube_prometheus: Update chart versions

NicolasT committed:
kube_nginx: Update chart version

NicolasT committed:
kube_heapster: Update chart version

NicolasT committed:
kube_metrics_server: Update chart version

Also update the `metrics-server` invocation arguments to support the new
daemon version and fix a small typo in a task name.

See: https://github.com/helm/charts/commit/bc1020abfd72d6062059ac51812b62ba80775008#diff-43ef3f69b43b1042c3b63d11277e466d

NicolasT committed:
cluster_logging: Update Elasticsearch chart version

Also set the `updateStrategy` of the `data` and `master` `StatefulSet`s
to `RollingUpdate`.

See: https://github.com/helm/charts/pull/7150
See: https://github.com/helm/charts/pull/7773

NicolasT committed:
cluster_logging: Update chart versions

bert-e commented:
# Hello NicolasT,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get
information on this process.





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @ssalaues

* @alxf

* @ballot-scality

* @Zempashi

* @NicolasT

* @anurag4DSB

* @thomasdanan

NicolasT committed:
docs: Update ChangeLog for #373

See: https://github.com/scality/metalk8s/pull/373

NicolasT commented:
To be investigated why the `metrics-server` test is failing.

NicolasT committed:
kube_metrics_server: Force Helm deployment

The default `state` value of `helm_cli` doesn't actually update a
release when it already exist. It needs to be forced to do so.

Fixes: 06cb37930d94c98a74c9c96bd2b7c8d5dedd1849
See: #372
See: https://github.com/scality/metalk8s/pull/372

NicolasT commented:
Tests were failing due to https://github.com/scality/metalk8s/pull/373/files#diff-36d18a34df637e8f16174465ed8e3350R8 not being there (DNS resolution of `ci-metalk8s` fails, of course).

NicolasT commented:
Ugh, new failure. Potentially timing issue: metrics-server doesn't immediately provide metrics, only after some time...

NicolasT commented:
Should be fixed... https://github.com/scality/metalk8s/pull/373/commits/69c8125afcc93fd82954d9b41188be21bc91ce55#diff-e7316cf09e38171b2f945d5fa5f92032R5

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.1`



The following branches will **NOT** be impacted:

* `development/1.0`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.

Zempashi commented on roles/kube_metrics_server/vars/main.yml:
```diff
@@ -2,6 +2,9 @@
+  - --logtostderr
+  # MetalK8s-specific:
+  - --kubelet-insecure-tls
+  # Required to make sure node hostnames aren't resolved through DNS
```
Should we adapt the inventory to not change the hostname of the server and get DNS working ?

NicolasT replied:
Then you rely on an external DNS server to resolve the hostname, which is not necessarily the case. Defaulting to InternalIP should be fine in most deployments, no? In the end, that's the IP `kubelet`s are reporting themselves.

Zempashi replied:
Yes, indeed. Now wondering why it's not the default

Zempashi replied:
maybe for tls...

NicolasT replied:
Kubespray not generating certificates correctly is why we need `--kubelet-insecure-tls` in the first place :-(

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.1`



The following branches have **NOT** changed:

* `development/1.0`



Please check the status of the associated issue None.

Goodbye NicolasT.
